torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -21,22 +21,29 @@ const inline char* getNcclErrorDetailStr(ncclResult_t error, c10::optional<std::
   if (processGroupFailureReason != c10::nullopt) {
     return (*processGroupFailureReason).c_str();
   }
+  std::string interpret = nullptr;
+  std::string err = std::string(ncclGetLastError(NULL));
```

Comment:
We can wrap the `ncclGetLastError()` in an if directive

Directive definition example: https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/NCCLUtils.hpp#L44-L49

Directive usage example:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/NCCLUtils.hpp#L115-L122